File: assignments/1251-Spring2025/assignment00-files/run_reports.py
"""
TODO: author/program documentation
"""
import datetime
import math
import pprint
import sys
import time

def seconds_between(a, b):
    """
    Computes and returns the difference in time in seconds between
    two given ISO8601 formatted strings `a` and `b`.
    """
    time_a = datetime.datetime.strptime(a, "%Y-%m-%dT%H:%M:%S")
    time_b = datetime.datetime.strptime(b, "%Y-%m-%dT%H:%M:%S")
    return (time_b-time_a).total_seconds()

def format_seconds(seconds):
    """
    Returns a formatted string of the given number of `seconds`
    into the format "HH:MM:SS"
    """
    return time.strftime('%H:%M:%S', time.gmtime(seconds))

File: assignments/1251-Spring2025/assignment00-files/RunReports.java
package unl.soc;

import java.time.LocalDateTime;
import java.time.temporal.ChronoUnit;

/**
 * TODO: Documentation, author, etc.
 */
public class RunReports {

	/**
	 * Returns the difference in time in *seconds* between two
	 * <code>LocalDateTime</code> instances. You may use, modify or adapt this
	 * method as you see fit.
	 *
	 * Returns a negative value if <code>b</code> represents a time *before*
	 * <code>a</code>.
	 *
	 * @param b
	 * @return
	 */
	public static int getTimeDiff(LocalDateTime a, LocalDateTime b) {
		return (int) ChronoUnit.SECONDS.between(a, b);
	}

	/**
	 * Returns a string formatted as "00:00:00" based on the given number of
	 * seconds. For example, if seconds = 4284, it would return "01:11:24" (1 hour,
	 * 11 minutes, 24 seconds).
	 *
	 * @param seconds
	 * @return
	 */
	public static String formatTime(int seconds) {
		int hours = seconds / 3600;
		seconds -= hours * 3600;
		int minutes = seconds / 60;
		seconds -= minutes * 60;
		return String.format("%02d:%02d:%02d", hours, minutes, seconds);
	}

	public static void main(String args[]) {
		//TODO: implement

		//an example of how to convert an ISO8601 string to a LocalDateTime:
		String s = "2024-07-27T10:36:48";
		LocalDateTime ldt = LocalDateTime.of(s);
	}

}
